src/renderers/WebGLRenderer.js

```diff
@@ -1821,6 +1821,10 @@ THREE.WebGLRenderer = function ( parameters ) {
 
 			return a.object.renderOrder - b.object.renderOrder;
 
+		} else if ( !!a.material.program && !!b.material.program && a.material.program.id !== b.material.program.id ) {
```

Comment:
````render -> opaqueObjects.sort( painterSortStable );````
happens just before ```render -> renderObject -> renderBuffer -> setProgram -> initMaterial``` which sets up the program so it isn't necessarily setup yet.